### ISSUE TITLE
[RFC] How to handle improper usage error

### ISSUE THREAD

**Issue by groyoh** [rails-api/active_model_serializers#1514]
In #1504, I figured out that in some part of the gem we do not make sure that user use the gem properly to match jsonapi specification.

The main example I ran into is when using a relationship that does not include `data` and do not use `links` and `meta` (Ref. https://github.com/rails-api/active_model_serializers/pull/1504#discussion_r52399639): 

``` ruby
has_one :author do
  include_data false
end
# => { relationships: { author: {} } }
```

This results in an empty relationship and is not allowed by jsonapi specification (http://jsonapi.org/format/#document-resource-object-relationships) but the gem does not prevent the user from doing so.

Should the gem raise an error in such case or let the user figure out on its own that he's using the gem wrongly?

**Comment by remear** [comment]
@groyoh Was this resolved by #1516?

**Comment by groyoh** [comment]
@remear no this was not resolved.
